### DISCUSSION TITLE
How can I make a floating window appear on top of a fullscreen window? (like `rofi`)

### DISCUSSION THREAD

**Discussion by ForeverNooob** [i3/i3#5263] [Q&A] [answered]
Hi, after a lot of mucking around (mostly by being confused, and failing, definitely failing), I've made a `dmenu_run` replacement which only relies on `fzf` and a terminal (in my case, that's Alacritty), as previously demonstrated [here](https://github.com/i3/i3/discussions/5250).

However... one of the most crucial features is being able to display this on top of fullscreen windows. Similar to `rofi`, when it is called it displays itself on top of any window.

I do have `popup_during_fullscreen leave_fullscreen` in my i3 config, but that's not exactly what I want. 
I want my floating window to be displayed on top of the fullscreen window without the fullscreen window leaving fullscreen.

So how can I do this in i3wm?

Thanks in advance.

**Comment by Airblader** [comment, marked as answer]
i3 doesn't allow any managed window on top of fullscreen windows. It works with rofi because rofi isn't managed by the window manager (it has override_redirect set).

**Reply by ForeverNooob** [reply]
Thanks, but is it even possible to make a window float on top of a fullscreen window without explicitly stating that in its sourcecode like `rofi` does?

I've managed to find out how I can set the [Override Redirect](https://tronche.com/gui/x/xlib/window/attributes/override-redirect.html) flag by doing:

`xdotool search --classname "fzf_launcher" set_window --overrideredirect 1 windowunmap windowmap`

Unfortunately using this method I can't seem to make the window float an top of a fullscreen window, and the window also stops receiving keyboard input. So I'm going to muck around some more and try to make this work if it's possible. If it's not, then there's not really much point in investing my time into this unless I convince Alacritty devs to add the same feature that `rofi` has.

**Reply by Airblader** [reply]
It's possible as you discovered, but generally not a good idea if the client isn't prepared to be unmanaged as that comes with a few responsibilities (such as input handling).

My answer would be "no it's not possible". It's not technically correct, but correct enough unless you really do want to get technical.
